### DISCUSSION TITLE
Any idea how to handle template system with grapejs?

### DISCUSSION THREAD

**Discussion by forrackun** [GrapesJS/grapesjs#4845] [Q&A] [answered]
if i am using template system like golang `html/template` which could contain conditional statements and variables if i want to to make them dynamic, any way i could make grapejs understand it 

```
<table>
    {{range .Employees}}
  <tr>
  <td>{{.EmployeeName}}</td>
  <td>{{.EmployeeSalary}}</td>
  <td>{{.EmployeeAge}}</td>
    <td>{{if .Done}}
            <span class="success">Active</span>
        {{else}}
            <span class="danger">Not Active</span>
        {{end}}
    </td>
  </tr>
    {{end}}
</table>
```

**Comment by artf** [comment, marked as answer]
yes it's possible but you'd need to properly create your [custom component](https://grapesjs.com/docs/modules/Components.html#define-custom-component-type) which resolves parsed HTML to components (I'd also suggest checking this https://github.com/artf/grapesjs/issues/4654)

**Reply by forrackun** [reply]
Nice, looks like much more involved than i imagined.

just to confirm [this](https://camo.githubusercontent.com/cf12576d282807a0330d9f93eceee06ee9b0f95dd20ad047e4668ec12376655c/687474703a2f2f6772617065736a732e636f6d2f696d672f676a732d636f6e636570742e706e67)
image suggest maybe grapejs has some why to achieve templated rendering but it is example of how it would potentially used not core functionality of libary and it is previewing much simpler form of templating system where fragments are yeield to base template file and would not support complex conditionals and loop constructs right ??

also, if one were to implement custom component would template system like alpinejs be more native to html (which has attributes that implements templating functions ) than golang/html since grapejs could just ignore attributes and render fragments inside as is it ??

thanks for quick reply

**Reply by artf** [reply]
Yeah there is no template system in the core of the framework, that image only suggest how you could manage different templates generated from the editor (eg. the main template could have some kind of text placeholders and then replaced with another template server-side). 

> also, if one were to implement custom component would template system like alpinejs be more native to html (which has attributes that implements templating functions ) than golang/html since grapejs could just ignore attributes and render fragments inside as is it ??

Yeah being more close to HTML it would be probably easier to manage.
